src/rust/engine/src/externs/interface.rs

```diff
@@ -1467,6 +1494,7 @@ fn ensure_remote_has_recursive(
   })
 }
 
+//Assumes a digest that comes from a snapshot - i.e. it is a directory with a single file in it
```

Comment:
What would be the best way to enforce this through code? It looks like at this point, we can't do any validation in Rust. The only call site appears to be `steraming_workunit_handler.py`, so I'm wondering if we can do eager validation in any functions leading up to this.

It'd be good to rename this to `directory_digests_with_single_files_to_bytes`. Horribly wordy, but that's our only defense here afaict to make sure the input is valid.

Also probably worth promoting to a doc comment `///`.